src/Hl7.Fhir.Support.Poco/Specification/Source/NpmPackageResolver.cs

```diff
@@ -0,0 +1,199 @@
+﻿using Firely.Fhir.Packages;
+using Hl7.Fhir.ElementModel;
+using Hl7.Fhir.Introspection;
+using Hl7.Fhir.Model;
+using Hl7.Fhir.Serialization;
+using Hl7.Fhir.Utility;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+#nullable enable
+
+namespace Hl7.Fhir.Specification.Source
+{
+    /// <summary>Reads FHIR artifacts (Profiles, ValueSets, ...) from one or multiple FHIR packages.</summary>
+    public class NpmPackageResolver : IAsyncResourceResolver, IArtifactSource
+    {
+        private PackageContext _context;
+        private ModelInspector _provider;
+
+        /// <summary>Create a new <see cref="NpmPackageResolver"/> instance to read FHIR artifacts from one or multiple FHIR packages of a specific FHIR version
+        /// found in the paths passed to this function.</summary>
+        /// <returns>A new <see cref="NpmPackageResolver"/> instance.</returns>
+        /// <param name="provider">A <see cref="ModelInspector"/> used to parse the filecontents to FHIR resources, this is typically a <see cref="ModelInspector"/> containing the definitions of a specific FHIR version. </param>
+        /// <param name="filePaths">A path to the FHIR package files.</param>
+        public NpmPackageResolver(ModelInspector provider, params string[] filePaths)
+        {
+            _context = TaskHelper.Await(() => createPackageContextAsync(filePaths));
```

Comment:
Would it not be better to make this a Lazy initialization? Constructors don't have to do the heavy lifting themselves.